freelist.go

```diff
@@ -249,24 +289,25 @@ func (f *freelist) read(p *page) {
 		f.ids = nil
 	} else {
 		ids := ((*[maxAllocSize]pgid)(unsafe.Pointer(&p.ptr)))[idx : idx+count]
-		f.ids = make([]pgid, len(ids))
-		copy(f.ids, ids)
+		f.readIDs(ids)
+	}
+}
 
-		// Make sure they're sorted.
-		sort.Sort(pgids(f.ids))
+// arrayReadIDs initializes the freelist from a given list of ids.
+func (f *freelist) arrayReadIDs(ids []pgid) {
+	if ids == nil {
+		return
 	}
+	f.ids = make([]pgid, len(ids))
+	copy(f.ids, ids)
```

Comment:
this change seems to be bad. now every readids call has to copy the array.

Reply:
ok